### ISSUE TITLE
[Feature] Align Radio widgets Options to the grid

### ISSUE THREAD

**Issue by hetunandu** [appsmithorg/appsmith#2072]
## Summary
Radio options are not aligned to the grid or aligning other elements around it is hard.

## Motivation

If our widgets hard equal spacing as the main grid, it would be easy to make alignment of UI

## Additional Context

![image](https://user-images.githubusercontent.com/12022471/101436682-aa1c5c00-3934-11eb-93ce-bf3c40138ca0.png)

**Comment by somangshu** [comment]
Allowing the options to grow in grid space as we add more options can be complex and involve too many calculations which result in a decrease in performance. 

@marks0351 Please correct me if I am wrong here. 

Although I see new problems here.

Today the option go outside the widget boundary:
![Screenshot 2021-05-14 at 11.35.47 AM.png](https://images.zenhubusercontent.com/60583007e1e4e5398dc9ec58/98564701-93b4-4602-9f16-6c6fe4fef84b)

And if there is another widget under the option widget, it flickers on mouse hover
https://images.zenhubusercontent.com/275993885/b6d94f8f-c94f-4c20-bf45-cf8489407417/screen_recording_2021_05_14_at_11_36_21_am.mov

**Comment by marks0351** [comment]
I think the issue is to align the options with the grid space. The widget is aligned but the options are not aligned to the rows of the canvas. Widget growth on addition of new options is not a good way to go coz it will have ripple effects on the canvas. The widget should always react within the space allocated to it. But we can have an attribute for all widgets to be flexible and react to the space allocated or be strict and only follow the grid spacing  and hide overflowing content. I think the reactive environment within widgets that we are planning to execute shud take care of this.

Any thoughts? @riodeuno @hetunandu

**Comment by hetunandu** [comment]
@somangshu @marks0351 @riodeuno I think the simple solution here is to ensure each option with margin is equal to our constant row height. We should not add extra options since that is a user configuration. This should ensure alignment of options with other widgets on the grid

**Comment by somangshu** [comment]
Got it. So we align the options to the `Row height/Grid space` (this is including the margin). And introduce a `scroll` property for the overflowing options? Does that make sense?

**Comment by riodeuno** [comment]
@somangshu @hetunandu @marks0351 If we restrict each entry to one row, what happens when we have multiline labels in one or more of the radio button options?

**Comment by hetunandu** [comment]
@riodeuno @somangshu @marks0351 This might no longer be a problem since we have updated the grid size to be more granular. Can we test this out? I see one problem that I cannot reduce the height of the text widget after a point

**Comment by riodeuno** [comment]
@hetunandu Yes we have minimum rows and columns value today which is closer to the old minimum of 1 row and column.
